dist/app.js

```diff
@@ -43,38 +43,70 @@ function setEvents () {
     let jump1 = pieces.getPieceFromArray($(`[x=${x+2}][y=${y+(2*number)}]`));
     let jump2 = pieces.getPieceFromArray($(`[x=${x-2}][y=${y+(2*number)}]`));
 
-    if (move1 === null || move2 === null){
+    if (move1 === null){
       console.log("can move")
       currentPiece.canMove = true;
+      currentPiece.validMoves.push({x: x + 1, y: y+number})
+      console.log(currentPiece)
+    }
+    if (move2 === null){
+      console.log("can move")
+      currentPiece.canMove = true;
+      currentPiece.validMoves.push({x: x - 1, y: y+number})
+      console.log(currentPiece)
     }
     if (move1){
       if (move1.color === opponent && !jump1){
         console.log("can jump")
         currentPiece.canJump = true;
+        currentPiece.canMove = false;
+        currentPiece.validJumps.push({x: x + 2, y: y+(2 * number)})
+        console.log(currentPiece)
       }
     }
     if (move2){
       if (move2.color === opponent && !jump2){
         console.log("can jump")
         currentPiece.canJump = true;
+        currentPiece.canMove = false;
+        currentPiece.validJumps.push({x: x - 2, y: y+(2 * number)})
+        console.log(currentPiece)
       }
     }
     _movePhase = currentPiece.canMove || currentPiece.canJump
   };
 
+//Okay for some reason currentPiece.validMoves grows out of control sometimes.
```

Comment:
This bug appears to be happening when you select a piece on the board where it does not have a valid jump forward-left (if a red piece on the left-hand side of the board), or a valid jump forward-right (if a red piece on the right-hand side of the board) or any move that would theoretically jump the piece off the board.

![image](https://cloud.githubusercontent.com/assets/11802078/17878484/afac366c-68b0-11e6-93d4-c3043bca7593.png)